From 9882d052190bcc76e1a175378f74d53f3f7a2a33 Mon Sep 17 00:00:00 2001
From: David King <amigadave@amigadave.com>
Date: Mon, 28 Sep 2015 16:04:48 +0100
Subject: Install D-Bus service file

Without this, it is impossible to launch the application from the
desktop file.

https://bugzilla.redhat.com/show_bug.cgi?id=1266868
https://bugzilla.gnome.org/show_bug.cgi?id=755746
---
 configure.ac                   |  1 +
 data/Makefile.am               | 10 +++++++++-
 data/org.gnome.Todo.service.in |  3 +++
 3 files changed, 13 insertions(+), 1 deletion(-)
 create mode 100644 data/org.gnome.Todo.service.in

diff --git a/configure.ac b/configure.ac
index bff6768..da0ad61 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,6 +29,7 @@ AX_COMPILER_FLAGS([GNOME_TODO_WARN_CFLAGS], [GNOME_TODO_WARN_LDFLAGS])
 
 AC_PROG_CC_C_O
 AC_PROG_INSTALL
+AC_PROG_SED
 
 # enable libtool
 LT_PREREQ([2.2.6])
diff --git a/data/Makefile.am b/data/Makefile.am
index 4a61434..3b70c21 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -18,6 +18,13 @@ gsettings_SCHEMAS = org.gnome.todo.gschema.xml
 
 @GSETTINGS_RULES@
 
+%.service: %.service.in
+	$(AM_V_GEN)$(SED) -e "s|[@]bindir[@]|$(bindir)|" $< > $@
+
+servicedir = $(datadir)/dbus-1/services
+service_DATA = $(service_in_files:.service.in=.service)
+service_in_files = org.gnome.Todo.service.in
+
 EXTRA_DIST=                     \
   appdata/org.gnome.Todo.appdata.xml \
   org.gnome.Todo.desktop \
@@ -39,7 +46,8 @@ EXTRA_DIST=                     \
   theme/bg.svg \
   $(appdata_in_files) \
   $(desktop_in_files) \
-  $(gsettings_SCHEMAS)
+  $(gsettings_SCHEMAS) \
+  $(service_in_files)
 
 CLEANFILES =                    \
   $(service_DATA)               \
diff --git a/data/org.gnome.Todo.service.in b/data/org.gnome.Todo.service.in
new file mode 100644
index 0000000..d0f7613
--- /dev/null
+++ b/data/org.gnome.Todo.service.in
@@ -0,0 +1,3 @@
+[D-BUS Service]
+Name=org.gnome.Todo
+Exec=@bindir@/gnome-todo --gapplication-service
-- 
cgit v0.11.2


